src/tests/unit/tests/reports/components/new-tab-link-confirmation-dialog.test.tsx

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { NewTabLink } from 'common/components/new-tab-link';
+import { shallow } from 'enzyme';
+import { isFunction } from 'lodash';
+import * as React from 'react';
+import {
+    ConfirmType,
+    NewTabLinkWithConfirmationDialog,
+} from 'reports/components/new-tab-link-confirmation-dialog';
+import { IMock, It, Mock, MockBehavior, Times } from 'typemoq';
+
+describe('NewTabLinkWithConfirmationDialog', () => {
+    it('generates unique id for each link', () => {
+        const testSubject1 = shallow(<NewTabLinkWithConfirmationDialog />);
+        const testSubject2 = shallow(<NewTabLinkWithConfirmationDialog />);
+
+        expect(testSubject1.find(NewTabLink).prop('id')).not.toEqual(
+            testSubject2.find(NewTabLink).prop('id'),
+        );
+    });
+
+    it('contains NewTabLink with forwarded props', () => {
+        const props = {
+            href: 'test-href',
+            title: 'test-title',
+        };
+        const child = <div>test-child</div>;
+
+        const testSubject = shallow(
+            <NewTabLinkWithConfirmationDialog {...props}>{child}</NewTabLinkWithConfirmationDialog>,
+        );
+
+        const testSubjectNewTabLink = testSubject.find(NewTabLink);
+
+        expect(testSubjectNewTabLink.prop('href')).toEqual(props.href);
```

Comment:
Would it be better to check for all props except the id here? Something like
```typescript
const { id, ...testableProps } = testSubjectNewTabLink.props(); // this "discards" the id from the props, but keeps all the others

expect(testableProps).toMatchSnapshot(); // in-code assertion should work just fine too
```

Reply:
Updated